### ISSUE TITLE
Training RT-DETRv2 and RT-DETRv4 on the tiny_coco dataset failed

### ISSUE THREAD

**Issue by wenhwu** [RT-DETRs/RT-DETRv4#13]
@Zhao-Yian , thank you for your excellent work.

When training `RT-DETRv2` and `RT-DETRv4` using the [tiny_coco_dataset](https://github.com/lizhogn/tiny_coco_dataset), the validation accuracy remained `0.0`, with results similar to the following:

```
Epoch: [71] Total time: 0:00:02 (0.7244 s / it)
Test:  [0/4]  eta: 0:00:02    time: 0.5399  data: 0.3519  max mem: 14933
Test:  [3/4]  eta: 0:00:00    time: 0.2889  data: 0.0958  max mem: 14933
Test: Total time: 0:00:01 (0.2990 s / it)
Averaged stats: 
Accumulating evaluation results...
COCOeval_opt.accumulate() finished...
DONE (t=0.02s).
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.002
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
best_stat: {'epoch': 47, 'coco_eval_bbox': 1.2425941389319655e-06}
```

I used the original configuration files `RT-DETRv4/configs/rtv2/rtv2_r101vd_6x_coco.yml` and `RT-DETRv4/configs/rtv4/rtv4_hgnetv2_x_coco.yml`, modifying only the number of GPUs from 4 to **2**, without changing any other hyperparameters.  the train log is [rt_detrv2_log.txt](https://github.com/user-attachments/files/23927321/rt_detrv2_log.txt) and [rt_detrv4_log.txt](https://github.com/user-attachments/files/23927324/rt_detrv4_log.txt).

What potential factors might be causing this training failure? thanks

**Comment by Zhao-Yian** [comment]
Based on the provided logs, the learning rate is very low, indicating that the number of training steps is very small, resulting in the model being barely updated. Could you please provide the amount of training data?

**Comment by mo-lx** [comment]
May I ask if it has been resolved? I have approximately 1,000 images in my dataset, and I also encountered the situation where the parameters hardly updated after many rounds of training, and the accuracy was very low.


Epoch: [12] Total time: 0:01:54 (6.7169 s / it)
Epoch 12: avg encoder grad 0.00% | distill 20.000000 -> 20.000000 (reset_to_default_zero_grad)
Test:  [0/3]  eta: 0:01:04    time: 21.4351  data: 15.1785  max mem: 20226
Test:  [2/3]  eta: 0:00:09    time: 9.4468  data: 5.0812  max mem: 20226
Test: Total time: 0:00:29 (9.7297 s / it)
Averaged stats: 
Accumulating evaluation results...
COCOeval_opt.accumulate() finished...
DONE (t=0.02s).
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.010
 Average Recall     (AR) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.007
 Average Recall     (AR) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
best_stat: {'epoch': 9, 'coco_eval_bbox': `2.0145681372777696e-05}
